lucetc/src/compiler.rs

```diff
@@ -47,6 +47,91 @@ impl OptLevel {
     }
 }
 
+pub struct CompilerBuilder {
+    pub target: Triple,
+    opt_level: OptLevel,
+    cpu_features: CpuFeatures,
+    heap_settings: HeapSettings,
+    count_instructions: bool,
+    canonicalize_nans: bool,
+    validator: Option<Validator>,
+}
+
+impl CompilerBuilder {
+    pub fn new() -> Self {
+        Self {
+            target: Triple::host(),
+            opt_level: OptLevel::default(),
+            cpu_features: CpuFeatures::default(),
+            heap_settings: HeapSettings::default(),
+            count_instructions: false,
+            canonicalize_nans: false,
+            validator: None,
+        }
+    }
+
+    pub fn target(&mut self, target: Triple) -> &mut Self {
+        self.target = target;
+        self
+    }
+
+    pub fn opt_level(&mut self, opt_level: OptLevel) -> &mut Self {
+        self.opt_level = opt_level;
+        self
+    }
+
+    pub fn cpu_features(&mut self, cpu_features: CpuFeatures) -> &mut Self {
+        self.cpu_features = cpu_features;
+        self
+    }
+
+    pub fn cpu_features_mut(&mut self) -> &mut CpuFeatures {
+        &mut self.cpu_features
+    }
+
+    pub fn heap_settings(&mut self, heap_settings: HeapSettings) -> &mut Self {
+        self.heap_settings = heap_settings;
+        self
+    }
+
+    pub fn heap_settings_mut(&mut self) -> &mut HeapSettings {
+        &mut self.heap_settings
+    }
+
+    pub fn count_instructions(&mut self, count_instructions: bool) -> &mut Self {
+        self.count_instructions = count_instructions;
+        self
+    }
+
+    pub fn canonicalize_nans(&mut self, canonicalize_nans: bool) -> &mut Self {
+        self.canonicalize_nans = canonicalize_nans;
+        self
+    }
+
+    pub fn validator(&mut self, validator: &Option<Validator>) -> &mut Self {
+        self.validator = validator.clone();
```

Comment:
Do you think that this could accept `Option<Validator>` instead? This will just move the `.clone()`'s to the places that this method is called.